doc/user_guide/cluster.rst

```diff
@@ -0,0 +1,152 @@
+
+Cluster analysis
+================
+
+.. versionadded:: 1.6
+
+A Hyperspy signal can represent a number of large arrays of different signals such as a map of chemical composition (EDS, XRF),
```

Comment:
I find that the documentation in the User Guide is too X-ray-centric. That would be fine for a demo or for a X-ray package, but in HyperSpy we should try to be a general as possible. I know that other parts of the User Guide suffer from the same issue, but that's for historical reasons and it is something that we have to fix.

I would suggest:

* Remove the xray and EM bias as much as possible
* Give a very brief description of what cluster analysis is and refer to a general review article for more information or even a wikipedia article.
* mention that HyperSpy basically wraps sklearn features and refer to their documentation for technical details
* focus on describing the logic of clustering as implemented here i.e. cluster on the navigation dimension or (if you take my suggestions elsewhere into account) "maps" obtained by e.g. curve fitting for labeling and use signal dimension in those regions to obtain the "centers".
* It would be good to highlight the parallelism between labels and loadings, and centers and factors.
* I would transform sentences advising the users how to process their data such as "Cluster analysis can be performed on raw data but is more commonly used on an extracted set of features." into descriptive ones e.g.  "Cluster analysis can be performed on raw data or on an extracted set of features.". In the end this guide should describe the features as implemented in HyperSpy, not explain how to process a certain kind of data, for that we have the demos.

Reply:
Removed the xray and EM bias as much as possible
Brief description and link to wikipedia article in introduction
Point to sklearn methods with links for technical details.
I think I've addressed the comments on format and content.